controlplane/kubeadm/internal/workload_cluster.go

```diff
@@ -53,12 +54,40 @@ var (
 	ErrControlPlaneMinNodes = errors.New("cluster has fewer than 2 control plane nodes; removing an etcd member is not supported")
 )
 
+// Common control-plane pod name prefixes
+const (
+	KubeAPIServerPodNamePrefix         = "kube-apiserver"
+	KubeControllerManagerPodNamePrefix = "kube-controller-manager"
+	KubeSchedulerHealthyPodNamePrefix  = "kube-scheduler"
+	EtcdPodNamePrefix                  = "etcd"
+)
+
+type ConditionReason string
+
+// GetSeverity returns the default severity for the condition reasons for the conditions owned by KCP.
+func (c ConditionReason) GetSeverity() clusterv1.ConditionSeverity {
+	switch {
+	case c == ConditionReason(clusterv1.PodProvisioningReason):
+		return clusterv1.ConditionSeverityInfo
+	case c == ConditionReason(clusterv1.PodMissingReason):
+		return clusterv1.ConditionSeverityWarning
+	case c == ConditionReason(clusterv1.PodFailedReason):
+		return clusterv1.ConditionSeverityError
+	case c == ConditionReason(clusterv1.EtcdMemberUnhealthyReason):
+		return clusterv1.ConditionSeverityError
+	case c == ConditionReason(controlplanev1.EtcdClusterUnhealthyReason):
+		return clusterv1.ConditionSeverityWarning
```

Comment:
```suggestion
	switch c {
	case ConditionReason(clusterv1.PodProvisioningReason):
		return clusterv1.ConditionSeverityInfo
	case ConditionReason(clusterv1.PodMissingReason):
		return clusterv1.ConditionSeverityWarning
	case ConditionReason(clusterv1.PodFailedReason):
		return clusterv1.ConditionSeverityError
	case ConditionReason(clusterv1.EtcdMemberUnhealthyReason):
		return clusterv1.ConditionSeverityError
	case ConditionReason(controlplanev1.EtcdClusterUnhealthyReason):
		return clusterv1.ConditionSeverityWarning
```

This should work I think?